map/src/components/map.tsx

```diff
@@ -1,20 +1,300 @@
 import React from 'react';
 import styled from 'styled-components';
-import { Filter } from 'src/data';
+import {Filter} from 'src/data';
+import {MARKERS} from "../data/markers";
 
 interface Props {
   className?: string;
   filter: Filter;
 }
 
-const Map = (props: Props) => {
-  const {className, filter} = props;
-  return (
-    <div className={className}>
-      MAP Filter:
-      {filter.service}
-    </div>
-  );
+class Map extends React.Component<Props, {}> {
+  private readonly googleMapRef: React.RefObject<any>;
+
+  private map: any;
+
+  private markers: any[] = [];
+
+  private serviceCircles: any[] = [];
+
+  private markerCluster: any = null;
+
+  private visibleMarkers: any = [];
+
+  private infoWindow: any = null;
+
+  private activeMarker: any = null;
+
+
+  constructor(props: Props) {
+    super(props);
+    this.googleMapRef = React.createRef();
+  }
+
+  componentDidMount(): void {
+
+    const googleMapScript = document.createElement('script');
+    googleMapScript.src = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyD5ywRBOAoyjwic5SzT9q3MPjLT1aibHO8&libraries=places'
+
+    const clusterLibScript = document.createElement('script');
+    clusterLibScript.src = 'https://unpkg.com/@google/markerclustererplus@4.0.1/dist/markerclustererplus.min.js';
+
+    // Load the scripts in order
+    window.document.body.appendChild(clusterLibScript);
+    clusterLibScript.addEventListener('load', () => {
+      window.document.body.appendChild(googleMapScript);
+    });
+
+    googleMapScript.addEventListener('load', () => {
+      this.map = this.createGoogleMap();
+
+      // This also likely needs a separate service
```

Comment:
@s0 This is one section where in the POC I hosted this is working, but I don't really know how the same stuff would work here.